Mathlib/GroupTheory/GroupAction/FixedPoints.lean

```diff
@@ -0,0 +1,331 @@
+the complements of `MulAction.fixedBy` and `AddAction.fixedBy` respectively.
+
+## Main theorems
+
+* `MulAction.fixedBy_mul`, `MulAction.movedBy_mul` (and their `AddAction` equivalents),
+  for the different relationship between the `movedBy`/`fixedBy` sets of `g*h`.
+* `MulAction.fixedBy_conj` and `MulAction.movedBy_conj`: the pointwise group action on the sets
+  `fixedBy α g` and `movedBy α g` translates to the conjugation action on `g`.
+* `MulAction.smul_mem_of_movedBy_subset` shows that if a set `s` is a superset of
+  `movedBy α g`, then the group action of `g` cannot send elements of `s` outside of `s`.
+* `MulAction.not_commute_of_disjoint_movedBy_preimage` allows one to prove that two group elements
+  do not commute from the disjointness of the `movedBy` set and its image by the second group
+  element, which is a property used in the proof of Rubin's theorem.
+-/
+
+namespace MulAction
+open Pointwise
+
+universe u v
+variable {α : Type v}
+variable {G : Type u} [Group G] [MulAction G α]
+variable {M : Type u} [Monoid M] [MulAction M α]
+
+
+section FixedPoints
+
+variable (α) in
+/-- In a multiplicative group action, the points fixed by `g` are also fixed by `g⁻¹` -/
+@[to_additive "In an additive group action, the points fixed by `g` are also fixed by `g⁻¹`"]
+theorem fixedBy_eq_fixedBy_inv (g : G) : fixedBy α g = fixedBy α g⁻¹ := by
+  ext x
+  repeat rw [mem_fixedBy]
+  constructor
+  all_goals (intro gx_eq_x; nth_rw 1 [← gx_eq_x])
+  · exact inv_smul_smul g x
+  · rw [← mul_smul, mul_right_inv, one_smul]
+
+@[to_additive]
+theorem smul_mem_fixedBy_iff_mem_fixedBy {a : α} {g : G} :
+    g • a ∈ fixedBy α g ↔ a ∈ fixedBy α g := by
+  rw [mem_fixedBy, smul_left_cancel_iff]
+  rfl
+
+theorem minimalPeriod_eq_one_of_fixedBy {a : α} {g : G} (a_in_fixedBy : a ∈ fixedBy α g) :
+    Function.minimalPeriod (fun x => g • x) a = 1 := by
+  rw [Function.minimalPeriod_eq_one_iff_isFixedPt]
+  exact a_in_fixedBy
+
+section AddAction.minimalPeriod_eq_one_of_fixedBy
+variable {G : Type u} [AddGroup G] [AddAction G α]
+
+theorem _root_.AddAction.minimalPeriod_eq_one_of_fixedBy {a : α} {g : G}
+    (a_in_fixedBy : a ∈ AddAction.fixedBy α g) :
+    Function.minimalPeriod (fun x => g +ᵥ x) a = 1 := by
+  rw [Function.minimalPeriod_eq_one_iff_isFixedPt]
+  exact a_in_fixedBy
+
+end AddAction.minimalPeriod_eq_one_of_fixedBy
+
+attribute [to_additive existing AddAction.minimalPeriod_eq_one_of_fixedBy]
+  minimalPeriod_eq_one_of_fixedBy
+
+variable (α) in
+@[to_additive]
+theorem fixedBy_subset_fixedBy_zpow (g : G) (j : ℤ) :
+    fixedBy α g ⊆ fixedBy α (g ^ j) := by
+  intro a a_in_fixedBy
+  rw [mem_fixedBy, zpow_smul_eq_iff_minimalPeriod_dvd,
+    minimalPeriod_eq_one_of_fixedBy a_in_fixedBy]
+  rw [Nat.cast_one]
+  exact one_dvd j
+
+variable (M α) in
+@[to_additive (attr := simp)]
+theorem fixedBy_one_eq_univ :
+    fixedBy α (1 : M) = Set.univ := by
+  ext a
+  rw [mem_fixedBy, one_smul]
+  exact ⟨fun _ => Set.mem_univ a, fun _ => rfl⟩
```

Comment:
```suggestion
  simp
```
or
```suggestions
  simp_rw [mem_fixedBy, one_smul, Set.mem_univ]
```

Reply:
This one I'm unsure if I agree; I can just have `Set.eq_univ_iff_forall.mpr (fun a => one_smul M a)`, though